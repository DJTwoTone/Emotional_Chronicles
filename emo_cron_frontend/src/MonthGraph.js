import React, { useContext, useEffect, useState } from 'react';
import { LineChart, Line, XAxis, YAxis, Legend, ResponsiveContainer} from 'recharts';
import { DateTime } from 'luxon'

import formatAnalysedMonth from './hooks/formatAnalysedMonth'; 

import UserContext from './UserContext';


function MonthGraph() {

    const { loggedInUser } = useContext(UserContext);
    const [formattedData, setFormattedData] = useState([])

    // console.log(loggedInUser)
    
    console.log('loggedinuser in the monthgraph', loggedInUser)

    useEffect(() => {
        async function fetchData() {
            const data = await formatAnalysedMonth(loggedInUser.username, DateTime.now());
            console.log(data)
        }
        fetchData();
    })

    

    


    return (

        <p>Month graph</p>
        // <ResponsiveContainer width="100%" height="100%">


        //     {/* <LineChart */}
        //     {/* <p>{data}</p> */}
        // </ResponsiveContainer>
    )
}

export default MonthGraph;