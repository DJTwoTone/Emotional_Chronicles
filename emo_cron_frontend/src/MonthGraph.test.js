import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import MonthGraph from './MonthGraph';

it('should render', function() {
    render(
        <MemoryRouter>
            <MonthGraph />
        </MemoryRouter>
    );
})

it('should match the snapshot', function() {
    const { asFragment } = render(
        <MemoryRouter>
            <MonthGraph />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
})