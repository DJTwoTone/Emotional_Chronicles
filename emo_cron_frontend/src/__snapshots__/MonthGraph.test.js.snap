// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <p>
      Month graph
    </p>
    <div
      class="recharts-responsive-container"
      style="width: 100%; height: 400px;"
    >
      <div
        height="400"
        width="0"
      >
        <div
          class="recharts-wrapper"
          style="position: relative; cursor: default; width: 500px; height: 250px;"
        >
          <svg
            class="recharts-surface"
            height="250"
            type="natural"
            version="1.1"
            viewBox="0 0 500 250"
            width="500"
          >
            <defs>
              <clippath
                id="recharts14-clip"
              >
                <rect
                  height="210"
                  width="390"
                  x="80"
                  y="5"
                />
              </clippath>
            </defs>
            <g
              class="recharts-cartesian-grid"
            >
              <g
                class="recharts-cartesian-grid-horizontal"
              >
                <line
                  fill="none"
                  height="210"
                  offset="[object Object]"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="390"
                  x="80"
                  x1="80"
                  x2="470"
                  y="5"
                  y1="5"
                  y2="5"
                />
                <line
                  fill="none"
                  height="210"
                  offset="[object Object]"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="390"
                  x="80"
                  x1="80"
                  x2="470"
                  y="5"
                  y1="215"
                  y2="215"
                />
              </g>
              <g
                class="recharts-cartesian-grid-vertical"
              >
                <line
                  fill="none"
                  height="210"
                  offset="[object Object]"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="390"
                  x="80"
                  x1="80"
                  x2="80"
                  y="5"
                  y1="5"
                  y2="215"
                />
                <line
                  fill="none"
                  height="210"
                  offset="[object Object]"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="390"
                  x="80"
                  x1="470"
                  x2="470"
                  y="5"
                  y1="5"
                  y2="215"
                />
              </g>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
            >
              <line
                class="recharts-cartesian-axis-line"
                fill="none"
                height="30"
                orientation="bottom"
                stroke="#666"
                type="category"
                width="390"
                x="80"
                x1="80"
                x2="470"
                y="215"
                y1="215"
                y2="215"
              />
              <g
                class="recharts-cartesian-axis-ticks"
              >
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="30"
                    orientation="bottom"
                    stroke="#666"
                    type="category"
                    width="390"
                    x="80"
                    x1="80"
                    x2="80"
                    y="215"
                    y1="221"
                    y2="215"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="30"
                    orientation="bottom"
                    stroke="none"
                    text-anchor="middle"
                    type="category"
                    width="390"
                    x="80"
                    y="223"
                  >
                    <tspan
                      dy="0.71em"
                      x="80"
                    >
                      0
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="30"
                    orientation="bottom"
                    stroke="#666"
                    type="category"
                    width="390"
                    x="80"
                    x1="470"
                    x2="470"
                    y="215"
                    y1="221"
                    y2="215"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="30"
                    orientation="bottom"
                    stroke="none"
                    text-anchor="middle"
                    type="category"
                    width="390"
                    x="470"
                    y="223"
                  >
                    <tspan
                      dy="0.71em"
                      x="470"
                    >
                      auto
                    </tspan>
                  </text>
                </g>
              </g>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
            >
              <line
                class="recharts-cartesian-axis-line"
                fill="none"
                height="210"
                orientation="left"
                stroke="#666"
                type="number"
                width="60"
                x="20"
                x1="80"
                x2="80"
                y="5"
                y1="5"
                y2="215"
              />
              <g
                class="recharts-cartesian-axis-ticks"
              />
            </g>
          </svg>
          <div
            class="recharts-legend-wrapper"
            style="position: absolute; width: 450px; height: auto; left: 20px; bottom: 5px;"
          >
            <ul
              class="recharts-default-legend"
              style="padding: 0px; margin: 0px; text-align: center;"
            >
              <li
                class="recharts-legend-item legend-item-0"
                style="display: inline-block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  version="1.1"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <path
                    class="recharts-legend-icon"
                    d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
                    fill="none"
                    stroke="#DF4ADF"
                    stroke-width="4"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                  style="color: rgb(223, 74, 223);"
                >
                  disgust
                </span>
              </li>
              <li
                class="recharts-legend-item legend-item-1"
                style="display: inline-block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  version="1.1"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <path
                    class="recharts-legend-icon"
                    d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
                    fill="none"
                    stroke="#008108"
                    stroke-width="4"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                  style="color: rgb(0, 129, 8);"
                >
                  fear
                </span>
              </li>
              <li
                class="recharts-legend-item legend-item-2"
                style="display: inline-block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  version="1.1"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <path
                    class="recharts-legend-icon"
                    d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
                    fill="none"
                    stroke="#FFE953"
                    stroke-width="4"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                  style="color: rgb(255, 233, 83);"
                >
                  joy
                </span>
              </li>
              <li
                class="recharts-legend-item legend-item-3"
                style="display: inline-block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  version="1.1"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <path
                    class="recharts-legend-icon"
                    d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
                    fill="none"
                    stroke="#5050FF"
                    stroke-width="4"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                  style="color: rgb(80, 80, 255);"
                >
                  sadness
                </span>
              </li>
              <li
                class="recharts-legend-item legend-item-4"
                style="display: inline-block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  version="1.1"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <path
                    class="recharts-legend-icon"
                    d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
                    fill="none"
                    stroke="#018AE1"
                    stroke-width="4"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                  style="color: rgb(1, 138, 225);"
                >
                  surprise
                </span>
              </li>
              <li
                class="recharts-legend-item legend-item-5"
                style="display: inline-block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  version="1.1"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <path
                    class="recharts-legend-icon"
                    d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
                    fill="none"
                    stroke="#00B510"
                    stroke-width="4"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                  style="color: rgb(0, 181, 16);"
                >
                  trust
                </span>
              </li>
            </ul>
          </div>
          <div
            class="recharts-tooltip-wrapper"
            style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
          >
            <div
              class="recharts-default-tooltip"
              style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
            >
              <p
                class="recharts-tooltip-label"
                style="margin: 0px;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
